lib/predictions/pub_sub.ex

```diff
@@ -31,14 +34,13 @@ defmodule Predictions.PredictionsPubSub do
     )
   end
 
-  @spec subscribe(String.t()) :: [Prediction.t()]
-  @spec subscribe(String.t(), GenServer.server()) :: [Prediction.t()] | {:error, term()}
-  def subscribe(topic, server \\ __MODULE__) do
+  @impl Behaviour
+  def subscribe(topic) do
     case StreamTopic.new(topic) do
       %StreamTopic{} = stream_topic ->
         :ok = StreamTopic.start_streams(stream_topic)
 
-        {registry_key, predictions} = GenServer.call(server, {:subscribe, stream_topic})
+        {registry_key, predictions} = GenServer.call(__MODULE__, {:subscribe, stream_topic})
```

Comment:
question: why not leave `__MODULE__` as a default argument to this function? Then we can inject different dependencies to test if we'd like.

Reply:
Because we are moving away from DI and towards using Mox. We don't need to inject variables in dynamically. That is, it will *never* be called with a second argument.